reenable testing cudnn SDPA with PyTorch dev version / 2.4.0a0+

I have disabled a number of tests failing with 2.4.0a0+. I imagine it is a PyTorch thing, but I'm a bit concerned nonetheless.

`thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16`
and one `FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: [cudnn_frontend] Error: No execution plans built successfully.`.

While investigating the latter, on my local this seems flaky and prone to hang in the second of the three (`float16`, `bfloat16`, `float32`) tests.

---

The CI job for CUDA PyTorch main branch fails with CUDNN errors since today-ish (European time). Seems to fail 100%, not randomly.

e.g.
https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=205024&view=logs&j=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&t=97651ec4-0b0f-5455-bbb5-3c30427a0a7e&l=11885

I don't have an idea yet what happened.

```
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_nvfuser_cuda_float16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_nvfuser_cuda_float16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_cudnn_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_nvfuser_cuda_bfloat16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_cudnn_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_nvfuser_cuda_bfloat16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_vjp_correctness_cudnn_sdpa[bfloat16-may-cat-grad-qkv] - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_vjp_correctness_cudnn_sdpa[float16-may-cat-grad-qkv] - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_vjp_correctness_cudnn_sdpa[bfloat16-never-cat-grad-qkv] - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
FAILED thunder/tests/test_cudnn_executor.py::test_vjp_correctness_cudnn_sdpa[float16-never-cat-grad-qkv] - RuntimeError: cuDNN Frontend error: CUDNN_BACKEND_OPERATION: cudnnFinalize Failed cudnn_status: CUDNN_STATUS_BAD_PARAM
```



cc @borda

With the merged PyTorch PR, we traded the old errors for a number of vjp correctness errors about different results in
`thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16`
and one `FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: [cudnn_frontend] Error: No execution plans built successfully.`.

While investigating the latter, on my local this seems flaky and prone to hang in the second of the three (`float16`, `bfloat16`, `float32`) tests.

@t-vi cudnn spda is now disabled on pyt main altogether. Moreover, pytorch main has moved to 2.5 and this thunder CI is now running cudnn sdpa tests successfully today. ([Link to thunder CI logs](https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=205929&view=logs&jobId=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&j=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&t=97651ec4-0b0f-5455-bbb5-3c30427a0a7e)) ([Link to revert commit](https://github.com/pytorch/pytorch/pull/125343))

I think we can reenable sdpa tests for 2.4 too, in case those still affect CI. Or atleast remove those `pytest.skip_if` macros.

> a number of vjp correctness errors about different results in
thunder/tests/test_grad.py::test_vjp_correctness_sdpa_manual_grad_forward_scaled_dot_product_attention_torch_cuda_bfloat16
and one FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_grad_forward_scaled_dot_product_attention_torch_cuda_float16 - RuntimeError: cuDNN Frontend error: [cudnn_frontend] Error: No execution plans built successfully..

Can you please provide details about your environment? Mainly the:
- the gpu arch
- cudnn frontend version
- cudnn backend version

```
import cudnn
print(cudnn.__version__)
print(cudnn.backend_version_string())
```
The `RuntimeError: cuDNN Frontend error:` makes me think that this is an old version of thunder/cudnn-frontend. Currently, cudnnex is supposed to reject versions before 1.3 for cudnn frontend.

--- a/thunder/tests/opinfos.py
+++ b/thunder/tests/opinfos.py
@@ -7880,15 +7880,6 @@ grad_sdpa_opinfo = OpInfo(
     # NOTE: NotImplementedError: Could not run 'aten::_scaled_dot_product_efficient_attention' with arguments from the 'CPU' backend.
     # NOTE: NotImplementedError: Could not run 'aten::_scaled_dot_product_efficient_attention_backward' with arguments from the 'CPU' backend
     devicetypes=(devices.DeviceType.CUDA,),
-    test_directives=(
-        DecorateInfo(
-            pytest.mark.skip(reason="https://github.com/Lightning-AI/lightning-thunder/issues/567"),
-            "test_core_vs_torch_consistency",
-            dtypes=(datatypes.bfloat16, datatypes.float16, datatypes.float32),
-            devicetypes=(devices.DeviceType.CUDA,),
-            active_if=version_between(torch.__version__, min_ver="2.4.0a0", max_ver="2.4.0a99"),
-        ),
-    ),
 )
 nn_ops.append(grad_sdpa_opinfo)
 

--- a/thunder/tests/test_cudnn_executor.py
+++ b/thunder/tests/test_cudnn_executor.py
@@ -200,10 +200,6 @@ def test_cudnn_vs_torch_consistency(op, device, dtype, *_):
     LooseVersion(cudnn.backend_version_string()) < LooseVersion("8.9.5"),
     reason="cuDNN is required to be at least `8.9.5`",
 )
-@pytest.mark.skipif(
-    version_between(torch.__version__, min_ver="2.4.0a0", max_ver="2.4.0a99"),
-    reason="https://github.com/Lightning-AI/lightning-thunder/issues/567",
-)
 @pytest.mark.parametrize("may_cat_grad_qkv", (True, False), ids=("may-cat-grad-qkv", "never-cat-grad-qkv"))
 @pytest.mark.parametrize("dtype", grad_sdpa_cudnn_opinfo.dtypes(), ids=tuple(map(str, grad_sdpa_cudnn_opinfo.dtypes())))
 def test_vjp_correctness_cudnn_sdpa(dtype, may_cat_grad_qkv):

--- a/thunder/tests/test_grad.py
+++ b/thunder/tests/test_grad.py
@@ -541,21 +541,12 @@ def test_vjp_correctness_index_put_manual(op, device, dtype, executor, comp):
 
 # NOTE Scaled_Dot_Product_Efficient_Attention_Backward does not support fp64 dtypes
 # RuntimeError: Only fp32, half & bf16 supported at the moment
-@pytest.mark.skipif(
-    not version_between(torch.__version__, min_ver="2.4.0a0", max_ver="2.4.0a99"),
-    reason="https://github.com/Lightning-AI/lightning-thunder/issues/567",
-)
 @ops(
     (get_opinfo("grad_forward_scaled_dot_product_attention"),),
     supported_dtypes=(dtypes.float16, dtypes.bfloat16),
     supported_devicetypes=(devices.DeviceType.CUDA,),
 )
 def test_vjp_correctness_sdpa_manual(op, device, dtype, executor, comp):
-    if version_between(torch.__version__, min_ver="2.4.0a0", max_ver="2.4.0a99"):
-        raise pytest.skip(
-            "https://github.com/Lightning-AI/lightning-thunder/issues/567",
-        )
-
     for sample in op.sample_inputs(device, dtype, requires_grad=True):
         from thunder.executors.sdpaex import sdpa_ex
 

